src/Coven.Agents.OpenAI/OpenAIGatewayConnection.cs

```diff
@@ -15,13 +16,15 @@ internal sealed class OpenAIGatewayConnection(
     [FromKeyedServices("Coven.InternalOpenAIScrivener")] IScrivener<OpenAIEntry> journal,
     ILogger<OpenAIGatewayConnection> logger,
     OpenAIClient openAIClient,
-    ITransmuter<OpenAIEntry, ResponseItem?> entryToItem)
+    ITransmuter<OpenAIEntry, ResponseItem?> entryToItem,
+    AgentStreamingOptions streamingOptions)
 {
     private readonly OpenAIClientConfig _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
     private readonly IScrivener<OpenAIEntry> _journal = journal ?? throw new ArgumentNullException(nameof(journal));
     private readonly ILogger _logger = logger ?? throw new ArgumentNullException(nameof(logger));
     private readonly OpenAIResponseClient _client = openAIClient.GetOpenAIResponseClient(configuration.Model) ?? throw new ArgumentNullException(nameof(openAIClient));
     private readonly ITransmuter<OpenAIEntry, ResponseItem?> _entryToItem = entryToItem ?? throw new ArgumentNullException(nameof(entryToItem));
+    private readonly AgentStreamingOptions _streaming = streamingOptions ?? throw new ArgumentNullException(nameof(streamingOptions));
```

Comment:
I'm thinking that we should separate this out to a separate type. We can decide how things compose based on whether users want streaming or not.